builtin/logical/aws/path_roles.go

```diff
@@ -58,94 +85,342 @@ func pathRoles() *framework.Path {
 }
 
 func (b *backend) pathRoleList(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	entries, err := req.Storage.List(ctx, "policy/")
+	b.roleMutex.RLock()
+	defer b.roleMutex.RUnlock()
+	entries, err := req.Storage.List(ctx, "role/")
 	if err != nil {
 		return nil, err
 	}
-	return logical.ListResponse(entries), nil
-}
-
-func pathRolesDelete(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
-	err := req.Storage.Delete(ctx, "policy/"+d.Get("name").(string))
+	legacyEntries, err := req.Storage.List(ctx, "policy/")
 	if err != nil {
 		return nil, err
 	}
 
+	return logical.ListResponse(append(entries, legacyEntries...)), nil
+}
+
+func (b *backend) pathRolesDelete(ctx context.Context, req *logical.Request, d *framework.FieldData) (*logical.Response, error) {
```

Comment:
I notice that some of these `framework.OperationFunc` are methods of the back end, and some aren't. Some do need the `b` to do stuff, and others don't. Would it sense to have _all_ of them be methods of the back end  for consistency? Or to only attach them when they need something off the back end?